### DISCUSSION TITLE
Open-metadata - kafka - airflow integrations versioning issue for "apache-airflow-providers-apache-kafka"

### DISCUSSION THREAD

**Discussion by triveni-radhik** [apache/airflow#38320] [General] [answered]
### What do you see as an issue?

Is your feature request related to a problem? Please describe.
I have installed open metadata(v1.3.0) and for compatible version of airflow (2.5.0) so that my open metadata and airflow connection and plugins working fine now I have installed apache kafka and I want to connection between apache kafka and airflow using "apache-airflow-providers-apache-kafka" this package but this package required airflow 2.6.0 version and If I'm upgrading version of airflow then it's not working with open-metadata is there any solutions for this type of issue ?

https://airflow.apache.org/docs/apache-airflow-providers-apache-kafka/stable/index.html
https://docs.astronomer.io/learn/airflow-kafka

**Please provide me a document or steps how can i upgrade version of airflow which will be compatible with open-metadata.**

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
Yes. Upgrade Airflow. This is always good idea. Every version of Airflow has many bugfixes and this is entirely planned that if you want to use new features from new versions of providers, you should upgrade to newer version of Airflow. This is our policy specifically for cases like this one where we want to incentivise our users who want to use newer versions of our providers to also migrate to newer versions of Airlfow. maintaining compatibility with old versions of Airlfow cost us a lot and we prefer our users to bear the cost of upgrading Airflow (which also includes security fixes implemented after the old versions of Airflow have been released. 

This is planned, documented, and we expect users like you to upgrade Airflow in such cases. 

Highly recommended. 

Converting to a discussion if needed, but this is all exactly as expected.

**Comment by potiuk** [comment]
You can read more here https://github.com/apache/airflow/blob/main/PROVIDERS.rst#upgrading-minimum-supported-version-of-airflow

**Reply by potiuk** [reply]
I highly recommend upgrading to 2.8.4 (we are about to release it in a few days) - in about a month all the  new versions of all providers wil bump their minimum Airflow version to  2.7.0:

> For example this means that by default we upgrade the minimum version of Airflow supported by providers to 2.7.0 in the first Provider's release after 27th of April 2024. 27th of April 2023 is the date when the first PATCHLEVEL of 2.6 (2.6.0) has been released.

**Reply by triveni-radhik** [reply]
Thanks for updating but issue was if I'm updating airflow to latest version then open-metadata plugins for airflow is not  compatible with latest version.
check here open metadata docs. for airflow https://docs.open-metadata.org/v1.3.x/deployment/ingestion/openmetadata

**Comment by triveni-radhik** [comment]
@potiuk , Thanks for replay , 
But issue was if I'm updating airflow to latest version then open-metadata plugins for airflow is not compatible with latest version.
check here open metadata docs. for airflow https://docs.open-metadata.org/v1.3.x/deployment/ingestion/openmetadata

**Reply by potiuk** [reply]
Then you should raise the issue to open-metadata.

**Reply by potiuk** [reply, marked as answer]
It's them who are far behind unforunately -  and we cannot be responsible for their lagging behind. Open a high priority issue with them - simply. They also have conflicting versioning information - they say it's Airflow 2.3-2.5 and claim that version 1.1.* will support Airflow 2.6 and they seem to be on 1.3.

You might also consider switching to something else than open-metadata - they seem to far behing many other things - like python version 3.7 - 3.9 where Python 3.12 is out, Python 3.7 is EOL already and 3.10 was released 2.5 years ago. You will soon get many more conflicts if you stick with them.

Also if you REALLY want - it's all open-source, you can likely copy sources of Kafka Provider to your DAGs and use it from there and hope there are no incompatibilities. Not guaranteed to work, but if you have no other choice as you are held back by someone else, that might be best solution to you. And you are on your own to solve any priblems.

**Reply by potiuk** [reply]
It's just the matter who bears the cost of those incompatibilities - we decided not to, now it's beetwen you and open-metadata to agree who is bearing the cost.

**Comment by triveni-radhik** [comment]
Thanks a lot @potiuk , 
It's work for me. will find some alternative solutions on open-metadata part.
I'm closing the discuss it's help me a lot.
